_posts/2016-04-11-growing-developers-at-greenfield.md

```diff
@@ -0,0 +1,84 @@
+me to look back on but also as a way to pause and recognize my successes and
+reflect on what I could do better tomorrow. If you spend all your energy focused
+on what you don’t know, it’s easy to [downplay the progress you're
+making](http://larahogan.me/donuts/). But If you concentrate on the *process* of
+learning something new each day, those small gains will add up to big wins over
+time. [All great developers were beginners](https://changelog.com/189/) at some
+point, and they achieved developer greatness one day a time, just like you.
+
+### Start Before You're Ready
+
+There’s a common misconception about learning, which assumes that learning and
+doing happen in two discrete steps, one after the other. Beginners often fall
+victim to this way of thinking because it can feel safer to focus on reading
+books, working through tutorials, and conceptual learning without getting one’s
+hands dirty. The truth, however, is that we learn by doing, and as software
+developers we learn to code by writing code. Becoming a better software
+developer is an iterative process, much like software development itself. When
+you execute your ideas, you discover what you really know and what you need to
+understand better. Each time you build something new, you learn something new.
+And there will always be more to learn, so the idea that you need to master a
+given language or technology before you begin is just not true. Instead, choose
+a side project to work on and troubleshoot as you go. Read the documentation.
+Dive into the source code. Ask for help on Slack or Stack Overflow. [Start
+before you're
+ready](http://sethgodin.typepad.com/seths_blog/2016/03/show-your-work.html)
+because if you wait until you feel ready, you may be waiting a long time.
+
+### Share What You Know (And What You Don't)
+
+It’s often said that the best way to learn is to teach what you know, and I’ve
+found this to be very true in software development. Even if you’re new to
+development, there’s definitely someone out there right now who is struggling to
+work through something you’ve just figured out. Sharing that knowledge on Stack
+Overflow, on Twitter, in a blog post not only helps others but also reinforces
+your own understanding. I often find that trying to communicate what I’ve
+learned to someone else helps cement what I know, and it more often than not
+reveals gaps in my understanding that I can then go and fill in. That last part
+is key. You don’t need to wait until you’ve written perfect code before you show
+it to others. Showing your work in progress, asking questions, and getting
+feedback are all key to learning. Contributing to [open source
+software](https://blog.newrelic.com/2014/05/05/open-source_gettingstarted/) is
+an especially great way to show your work, help others, and learn while you do
+it. The best code is not written alone. Participating in the community — giving,
```

Comment:
I think this line "Contributing to open source . . . while you do it" is out of place. I would swap it with "The best code is not written alone." Maybe that flows better? I also think "The best code is not written alone" is a great line and could be the opening heading for this paragraph.